informal_update/utils.py

```diff
@@ -0,0 +1,90 @@
+    result = BytesIO()
+    pdf = pisa.pisaDocument(BytesIO(html.encode("ISO-8859-1")), result)
+    if not pdf.err:
+        file = {
+            'filename': 'informal_update.pdf',
+            'file': result.getvalue()
+        }
+        return file
+    return None
+
+
+def generate_file_data(data, list):
+    for file in data:
+        file_dict = dict()
+        if file['file']:
+            file_url = settings.BASE_URL + file['file']
+            file_dict['image'] = file_url
+        file_dict['caption'] = file['caption']
+        list.append(file_dict)
+
+
+def send_email_when_informal_update_created(instance):
+    from informal_update.serializers import InformalUpdateSerializer
+
+    informal_update = InformalUpdateSerializer(instance)
+    data = informal_update.data
+    map_list = []
+    graphics_list = []
+    generate_file_data(data['map_files'], map_list)
+    generate_file_data(data['graphics_files'], graphics_list)
+    email_context = {
+        'resource_url': get_project_url(instance.id),
+        'title': data['title'],
+        'situational_overview': data['situational_overview'],
+        'map_list': map_list,
+        'graphic_list': graphics_list,
+        'actions_taken': data['actions_taken'],
+        'resources': data['references'],
+    }
+    share_with_group = instance.share_with
+    if share_with_group:
+        if not share_with_group == InformalUpdate.InformalShareWith.RCRC_NETWORK_AND_DONOR:
+            email_subscription = InformalEmailSubscriptions.objects.get(
+                share_with=str(share_with_group)
+            )
+            group = email_subscription.group
+            if group:
+                users = User.objects.filter(groups=group)
+                users_emails = list(users.values_list('email', flat=True))
+                send_notification(
+                    'Informal Update',
+                    users_emails,
+                    render_to_string('email/informal_update/informal_update.html', email_context),
+                    'Informal Update'
+                )
+        else:
+            rcrc_users_email = list(
+                User.objects.filter(
+                    groups__name=InformalUpdate.InformalShareWith.RCRC_NETWORK
+                ).values_list('email', flat=True)
+            )
+            donors_email = list(Donors.objects.all().values_list('email', flat=True))
+            send_notification(
+                'Informal Update',
+                rcrc_users_email,
+                render_to_string('email/informal_update/informal_update.html', email_context),
+                'Informal Update'
+            )
+            send_notification(
+                'Informal Update',
+                donors_email,
+                render_to_string('email/informal_update/donor_email.html', email_context),
+                'Informal Update',
+                render_to_pdf('email/informal_update/informal_pdf.html', email_context)
+            )
+    return email_context
```

Comment:
This seems to be complicated, let's simplify this to something like this.
NOTE: Using `groups__informal_email_subscription__share_with` here in relation to `nformal_email_subscription` related_name in **InformalEmailSubscription**.
```suggestion
    share_with_group = instance.share_with
    if instance.share_with is None:
        # Nothing to do here
        return

    # Generate donors, users email through share_with_group config
    users_emails = []
    donors_emails = []
    if share_with_group == InformalUpdate.InformalShareWith.RCRC_NETWORK_AND_DONOR:
        # NOTE: Custom logic for RCRC_NETWORK_AND_DONOR, fetch donors and RCRC_NETWORK users
        users_emails = User.objects.filter(groups__informal_email_subscription__share_with=InformalUpdate.InformalShareWith.RCRC_NETWORK)
        donors_email = list(Donors.objects.all().values_list('email', flat=True))
    else:
        users_emails = User.objects.filter(groups__informal_email_subscription__share_with=share_with_group)

    # If there are recipients in the groups send the email
    if users_emails:
        send_notification(
            'Informal Update',
            users_emails,
            render_to_string('email/informal_update/informal_update.html', email_context),
            'Informal Update'
        )

    if donors_emails:
        send_notification(
            'Informal Update',
            donors_email,
            render_to_string('email/informal_update/donor_email.html', email_context),
            'Informal Update',
            render_to_pdf('email/informal_update/informal_pdf.html', email_context)
        )
```